### DISCUSSION TITLE
How secure is authwithpassword for users?

### DISCUSSION THREAD

**Discussion by MoneyDropLobby** [pocketbase/pocketbase#3997] [Q&A]
I am wondering if the request to the api with the password is encrypted to maintain security while using http

**Comment by ganigeorgiev** [comment]
I'm not sure that I understand the question.

Nothing is encrypted if you are sending HTTP requests.
The submitted request data will be encrypted only if you use HTTPS which is more-or-less a requirement with modern browsers.

**Comment by MoneyDropLobby** [comment]
Yeah I know but you should implement an encryption/decryption with a secret
phrase or sum.

**Reply by ganigeorgiev** [reply]
It doesn't make much sense to have a client request data encryption if you are going to use TLS anyway.

**Reply by ganigeorgiev** [reply]
Maybe you are confusing how the passwords are stored in the database?
If that is what you are asking, then we use bcrypt to hash the passwords.

**Reply by MoneyDropLobby** [reply]
Yeah that makes no sense but for those who use http only

**Reply by ganigeorgiev** [reply]
There are no plans for client request data encryption out of the box.
If you want your users to have a secure connection with your PocketBase server, the only reliable way is to use HTTPS.

**Comment by jonshipman** [comment]
What is your reason to use HTTP only? As far as I'm concerned, for the past decade HTTPS should be the default and for the past 5 years having HTTP enabled is silly. All internet traffic should be over HTTPS. Even little ole' me serving tiny little websites uses Caddy to reverseproxy and it uses Let's Encrypt OOTB on all domains. You have to pass `tls false` to even disable it.

Even if you're morally against the ISRG for some reason, you have the option to issue a self-cert. Though probably would only work in a corporate environment where IT could auto trust the certificate on all new deployments.
